python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -182,11 +190,36 @@ def my_asset(my_upstream_asset: int) -> int:
     """
 
     def create_asset():
+        if non_argument_deps is not None and deps is not None:
+            raise DagsterInvalidDefinitionError(
+                "Cannot specify both deps and non_argument_deps to @asset. Use only deps instead."
+            )
+
+        upstream_asset_deps: Optional[
+            Sequence[Union[CoercibleToAssetKey, AssetsDefinition, SourceAsset]]
+        ] = None
+        if deps is not None:
+            for dep in deps:
+                if isinstance(dep, AssetsDefinition) and len(dep.keys) > 1:
+                    raise DagsterInvalidDefinitionError(
+                        "Cannot pass a multi_asset AssetsDefinition as an argument to deps."
+                        " Instead, specify dependencies on the assets created by the multi_asset"
+                        f" via AssetKeys or strings. For the multi_asset {dep.node_def.name}, the"
+                        f" available keys are: {dep.keys}."
+                    )
+            upstream_asset_deps = deps
```

Comment:
type checks on deps? 

Need to make sure a CacheableAssetsDefinitiion isn't passed and if it is that we give a good error message